src/kaitai_spec/attribute/enumeration.rs

```diff
@@ -1,31 +1,56 @@
+use super::{Attribute, ContextNoContents, ContextTypeAttributes};
+use either::Either;
+use std::collections::HashMap;
+
 #[derive(Debug, PartialEq)]
 pub struct Enumeration {
     pub name: String,
-    pub type_: EnumType,
+    pub type_: super::common_types::IntType,
 }
 
-#[derive(Debug, PartialEq)]
-pub enum EnumType {
-    U1,
-    S1,
-    Long {
-        type_: LongType,
-        endian: Option<Endian>,
-    },
+pub fn try_build<'de, A>(
+    context: ContextNoContents,
+) -> Either<Result<Attribute, A::Error>, ContextTypeAttributes>
+where
+    A: serde::de::MapAccess<'de>,
+{
+    match context.type_attributes.is_some() {
```

Comment:
Please match the `type_attributes` field directly:
```rust
let enumeration = match context.type_attributes {
    Some(Either(Left(enumeration))) => enumeration,
    Some(Either(Right(value))) => return Either::Riget(ContextTypeAttributes { ... })),
    None => return ...,
};
```

and all the remaining logic directly in this func. Please also don't use `clone`, `unwrap`, `etc`